src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -57,19 +53,25 @@ public void PopulateFeature(IEnumerable<ApplicationPart> parts, ViewsFeature fea
         }
 
         /// <summary>
-        /// Gets the type of <see cref="ViewInfoContainer"/> for the specified <paramref name="assemblyPart"/>.
+        /// Gets the sequence of <see cref="RazorViewAttribute"/> instances associated with the specified <paramref name="assemblyPart"/>.
         /// </summary>
         /// <param name="assemblyPart">The <see cref="AssemblyPart"/>.</param>
-        /// <returns>The <see cref="ViewInfoContainer"/> <see cref="Type"/>.</returns>
-        protected virtual ViewInfoContainer GetManifest(AssemblyPart assemblyPart)
+        /// <returns>The sequence of <see cref="RazorViewAttribute"/> instances.</returns>
+        protected virtual IEnumerable<RazorViewAttribute> GetViewAttributes(AssemblyPart assemblyPart)
         {
-            var type = CompiledViewManfiest.GetManifestType(assemblyPart, FullyQualifiedManifestTypeName);
-            if (type != null)
+            if (assemblyPart == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyPart));
+            }
+
+            var featureAssembly = CompiledViewManfiest.GetFeatureAssembly(assemblyPart);
+            if (featureAssembly != null)
             {
-                return (ViewInfoContainer)Activator.CreateInstance(type);
+                return featureAssembly.GetCustomAttributes<RazorViewAttribute>()
+                    .Where(attribute => attribute.GetType() == typeof(RazorViewAttribute));
```

Comment:
Maybe not. We can remove the other page feature provider and let this type do the discovery for pages and views alike.